### PR TITLE
docs: add serialization section to data fetching

### PR DESCRIPTION
**Pull request by mahdiboomeri**
### ❓ Type of change

- [x] 📖 Documentation (updates to the documentation or readme)
- [ ] 🐞 Bug fix (a non-breaking change that fixes an issue)
- [ ] 👌 Enhancement (improving an existing functionality like performance)
- [ ] ✨ New feature (a non-breaking change that adds functionality)
- [ ] ⚠️ Breaking change (fix or feature that would cause existing functionality to change)

### 📚 Description

This PR adds a serialization section to the data fetching page, including information on type conversion behavior, `toJSON`, and a workaround for using superjson (and other serializers). 
follow-up to https://github.com/unjs/nitro/pull/1002.

### 📝 Checklist

- [x] I have linked an issue or discussion.
- [x] I have updated the documentation accordingly.

### CODE REVIEW

File: docs/1.getting-started/6.data-fetching.md [hunk 1]
```diff
@@ -352,3 +352,105 @@ Using `$fetch` has a number of benefits, including:
 It will handle 'smartly' making direct API calls if it's running on the server, or making a client-side call to your API if it's running on the client. (It can also handle calling third-party APIs.)
 
 Plus, it comes with convenience features including automatically parsing responses and stringifying data.
+
+## Serialization
+
+::Alert
+This section is not released yet. You can beta test using [Edge Channel](https://nuxt.com/docs/guide/going-further/edge-channel).
+::
+
```
Comment (danielroe) [inline docs/1.getting-started/6.data-fetching.md:361]:
```suggestion
```

File: docs/1.getting-started/6.data-fetching.md [hunk 2]
```diff
@@ -352,3 +352,105 @@ Using `$fetch` has a number of benefits, including:
...
+When fetching data from the `server` directory, the response is serialized using `JSON.stringify`. However, since serialization is limited to only JavaScript primitive types, Nuxt does its best to convert the return type of `$fetch` and `useFetch` to match the actual value.
+
+::alert{icon=👉}
+You can learn more about JSON.stringify limitations [here](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/JSON/stringify#description).
```
Comment (danielroe) [inline docs/1.getting-started/6.data-fetching.md:365]:
```suggestion
You can learn more about `JSON.stringify` limitations [here](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/JSON/stringify#description).
```

File: docs/1.getting-started/6.data-fetching.md [hunk 3]
```diff
@@ -352,3 +352,105 @@ Using `$fetch` has a number of benefits, including:
...
+::
+
+### Example
+
+```ts [server/api/foo.ts]
+export default defineEventHandler(() => {
+  return new Date();
+});
```
Comment (danielroe) [inline docs/1.getting-started/6.data-fetching.md:373]:
```suggestion
  return new Date()
})
```

File: docs/1.getting-started/6.data-fetching.md [hunk 4]
```diff
@@ -352,3 +352,105 @@ Using `$fetch` has a number of benefits, including:
...
+```
+
+```vue [app.vue]
+<script setup lang="ts">
+// Type of `data` is infered as string even though we returned a Date object
+const { data } = await useFetch("/api/foo");
```
Comment (danielroe) [inline docs/1.getting-started/6.data-fetching.md:379]:
```suggestion
const { data } = await useFetch("/api/foo")
```

File: docs/1.getting-started/6.data-fetching.md [hunk 5]
```diff
@@ -352,3 +352,105 @@ Using `$fetch` has a number of benefits, including:
...
+
+```vue [app.vue]
+<script setup lang="ts">
+// Type of `data` is infered as 
+// {
+//  createdAt: {
+//     year: number;
+//     month: number;
+//     day: number;
+//  }
```
Comment (danielroe) [inline docs/1.getting-started/6.data-fetching.md:415]:
```suggestion
//   createdAt: {
//     year: number;
//     month: number;
//     day: number;
//   }
```

File: docs/1.getting-started/6.data-fetching.md [hunk 6]
```diff
@@ -352,3 +352,105 @@ Using `$fetch` has a number of benefits, including:
...
+    },
+  };
+  return data;
+});
+
+```
+
+```vue [app.vue]
+<script setup lang="ts">
+// Type of `data` is infered as
```
Comment (danielroe) [inline docs/1.getting-started/6.data-fetching.md:409]:
```suggestion
// Type of `data` is inferred as
```

File: docs/1.getting-started/6.data-fetching.md [hunk 7]
```diff
@@ -352,3 +352,105 @@ Using `$fetch` has a number of benefits, including:
...
+//     day: number;
+//  }
+// }
+const { data } = await useFetch("/api/bar");
+</script>
+```
+
+### Using an alternative serializer
+
+Unfortunately, Nuxt does not currently support an alternative serializer to `JSON.stringify`. However, you can return your payload as a normal string and utilize the `toJSON` method to maintain type safety.
```
Comment (danielroe) [inline docs/1.getting-started/6.data-fetching.md:423]:
```suggestion
Nuxt does not currently support an alternative serializer to `JSON.stringify`. However, you can return your payload as a normal string and utilize the `toJSON` method to maintain type safety.
```

**Review by danielroe — APPROVED** [review]

**Review by pi0 — APPROVED** [review]
Nice additions 💯

**Comment by mahdiboomeri** [comment]
@danielroe Thanks for the review. I was checking the doc site again and superjson link is broken. I forgot to add https :)

**Comment by danielroe** [comment]
fixed ✅

thanks for spotting!

**Pull request merged**